src/app/lib/utilities/getStoryPromoInfo/index.jsx

```diff
@@ -28,10 +28,15 @@ export const getHeadline = item => {
   return headline;
 };
 
-export const getUrl = item => {
+export const getUrl = (item, service = null, variant = null) => {
   const assetUri = pathOr(null, ['locators', 'assetUri'], item);
   const canonicalUrl = pathOr(null, ['locators', 'canonicalUrl'], item);
-  const uri = pathOr(null, ['uri'], item);
+  let uri = pathOr(null, ['uri'], item);
+  if (uri && service === 'zhongwen') {
+    if (uri.indexOf(`/${variant}`) === -1 && uri.indexOf('/articles/') !== -1) {
+      uri = uri.substr(0, uri.length - variant.length) + variant;
+    }
```

Comment:
Could we use replace instead substring? (I suppose substring is arguably more secure?)

Reply:
I think the substring is agnostic of which service this is being applied on i.e. trad -> simp or cyr -> lat

A replace would need a specific string